test/Elastic.Apm.Tests/SpanTests.cs

```diff
@@ -0,0 +1,30 @@
+using Elastic.Apm.Tests.Mocks;
+using FluentAssertions;
+using Xunit;
+
+namespace Elastic.Apm.Tests
+{
+	public class SpanTests
+	{
+		[Fact]
+		public void End_ShouldRestoreParentSpan_WhenTransactionIsNotSampled()
+		{
+			// Arrange
+			var payloadSender = new MockPayloadSender();
+			using (var agent =
+				new ApmAgent(new TestAgentComponents(config: new MockConfigSnapshot(transactionSampleRate: "0"), payloadSender: payloadSender)))
+			{
+				var transaction = agent.Tracer.StartTransaction("transaction", "type");
+
+				var parentSpan = transaction.StartSpan("parent", "type");
+
+				// Act
+				parentSpan.CaptureSpan("span", "type", span => { });
+
+				// Assert
+				payloadSender.Spans.Count.Should().Be(0);
+				agent.TracerInternal.CurrentExecutionSegmentsContainer.CurrentSpan.Should().Be(parentSpan);

```

Comment:
```suggestion
				agent.Tracer.CurrentSpan.Should().Be(parentSpan);
```

I think `agent.Tracer.CurrentSpan` is preferable to `agent.TracerInternal.CurrentExecutionSegmentsContainer` since the former is public API.